src/Analyzer.JsonRuleEngine.UnitTests/AnyOfExpressionTests.cs

```diff
@@ -16,6 +16,23 @@ namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.UnitTests
     [TestClass]
     public class AnyOfExpressionTests
     {
+        /// <summary>
+        /// A mock implementation of an <see cref="Expression"/> for testing internal methods.
+        /// </summary>
+        private class MockExpression : Expression
```

Comment:
Maybe is worth moving this to its own file so there's no repeated code between AnyOfExpressionTests and AllOfExpressionTests?